Add generate_chat_logs script and sample

eric-wayne opened the pull request:
Allen Dean and German suggested this repo as a good place to publish a script that developers can use to simulate users and generate chat logs with Conversation. This PR updates the README.md to add a new section at the bottom explaining the generate_chat_logs.py3 python script and sample utterances CSV file.  It also includes those two new files.  Along the way, I fixed some typos in REAME.md, particularly around the anchor tags.

Note that Dan O'Connor and I are about to publish a complementary Jupyter notebook to IBM DSX Community. In that notebook we will refer to this repo and this script if they need help generating log data to use with the notebook. 

Thanks!

eric-wayne committed:
Update README.md

fixed 5-6 anchor tags in the contents. Fixed link to the cloud-functions json file.

eric-wayne committed:
Merge pull request #1 from eric-wayne/eric-wayne-patch-1

Update README.md

eric-wayne committed:
Update README.md

Fixed two typos with links

eric-wayne committed:
Update README.md

add section on generating

eric-wayne committed:
Update README.md

eric-wayne committed:
Update README.md

eric-wayne committed:
Update README.md

eric-wayne committed:
Update README.md

eric-wayne committed:
Update README.md

eric-wayne committed:
Update README.md

eric-wayne committed:
adding generate_chat_logs.py3 by upload

eric-wayne committed:
Adding by upload

Adding utterances-for-generate-chat-logs.csv by upload

eric-wayne committed:
Update README.md

eric-wayne committed:
Update README.md

eric-wayne committed:
Fix userid parameter in script

michelle-miller reviewed (approved):
looks good

germanattanasio commented on conversation/generate_chat_logs.py3:
```diff
@@ -0,0 +1,132 @@
+            turnCount = 0
+            continue
+
+        message = { "input": { "text": turn["question"] } ,"alternate_intents": True}
```
If you use the SDK you won't have worry about TLS2 or building the REST API on your own.
I imagine something like

```py
conversation = ConversationV1(
    username=os.getenv("CONVERSATION_USERNAME"),
    password=os.getenv("CONVERSATION_PASSWORD"),
    version='2017-04-21',
    default_headers= { 'X-Watson-Metadata' : 'customer_id=' + args.customer_id })


workspace_id = '506e4a2e-3d5d-4dca-b374-38edbb4139ab'
if os.getenv("conversation_workspace_id") is not None:
    workspace_id = os.getenv("conversation_workspace_id")

response = conversation.message(workspace_id=workspace_id, input={
    'text': 'What\'s the weather like?'})
print(json.dumps(response, indent=2))

germanattanasio reviewed (changes requested):
Please use the SDK and the `default_headers` don't try to write the API on your own

eric-wayne committed:
Update README.md

eric-wayne committed:
Delete generate_chat_logs.py3

eric-wayne committed:
Use rewritten script from German plus tweaks

Based on the PR changes requested and German's rewritten script.  German made a bunch of improvements, including supporting Python 2 and using the Conversation Python SDK.  I tweaked German's version in small ways including
- added more checks on input parameters. Made url optional with a default.
- removed the CSV answer regex and intents handling. Their origin was for testing while this script is focused on generation only.

eric-wayne committed:
Adjusted Sample CSV to match script

Now that the script is simplified and no longer handles answer regex, the header row of the CSV only includes questions. Also added a greeting utterance to each conversation sequence so that the actual user intended utterance is passed in the same conversation.  This allows dialog to be more properly triggered.

eric-wayne committed:
undo putting CSV in wrong place

eric-wayne committed:
Change CSV to match script

Now that the script is simplified and no longer handles answer regex, the header row of the CSV only includes questions. Also added a greeting utterance to each conversation sequence so that the actual user intended utterance is passed in the same conversation.  This allows dialog to be more properly triggered.

eric-wayne commented:
Closing this PR - significant changes coming. Will submit a new PR based on German's rewritten script.
